grade.sh

```
@@ -0,0 +1,198 @@
+  if [[ "$FAST" == true && "$fast_skip" == true ]]; then
+    CHECKS+=("{\"name\":\"$name\",\"status\":\"skipped\",\"score\":0,\"max\":$weight,\"detail\":\"skipped in fast mode\"}")
+    return 0
+  fi
+  
+  MAX=$((MAX + weight))
+  if eval "$cmd" 2>&1 | tee "$REPORT_DIR/${name}.log" >"$REPORT_DIR/${name}.raw" 2>&1; then
+    SCORE=$((SCORE + weight))
+    CHECKS+=("{\"name\":\"$name\",\"status\":\"pass\",\"score\":$weight,\"max\":$weight,\"detail\":\"\"}")
+    echo "  [PASS] $name"
+  else
+    local detail="$(head -5 "$REPORT_DIR/${name}.raw" | tr '\n' ' ')"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)